Release Note v11.4.0 (#393)
mia-platform/documentation: docs/release-notes/v11.4.0.md

---
id: v11.4.0
title: Version v11.4.0 Release Notes
sidebar_label: v11.4.0
image: "img/release-note-link-preview.png"
---

_July 06th, 2023_

## Console

### New Features

#### What's new section

A big welcome to the [What's new section](/development_suite/overview-dev-suite.md#whats-new), where you can stay up to date with the latest product news!  
You can access this section by clicking on the question mark icon located in the top bar of the Console interface.
In case there is any unread news, a notification symbol will appear, so that you do not miss the latest updates that can improve your experience on Mia-Platform Console!

#### Support to automatic logout from identity provider

> From now on, Self-Hosted Mia-Platform Console installations support the configuration of user automatic logout from an OIDC compliant identity provider when logging out from the Console itself. For more information, visit the related [Self-Hosted installations documentation page](/self_hosted/installation-chart/30_authentication_provider.md#logout-flow).

Reviewer: ```suggestion
From now on, Mia-Platform Console installations support the configuration of user automatic logout from an OIDC compliant identity provider when logging out from the Console itself. For more information, visit the related [Mia-Platform installations documentation page](/self_hosted/installation-chart/30_authentication_provider.md#logout-flow).
```

Reviewer: should we remove also the link to the self hosted installation documentation page?

#### Index autogeneration for projections creates indexes for `__internal_counter` and `__internal_counterType`

The Console can now automatically generate the `__internal_counter` and `__internal_counterType` indexes, which have been introduced from `v7.5.0` of the Real-Time Updater in the update Projection queries.

#### Aggregation ConfigMap automatically handled inside the Single View Creator microservice

When attaching a Single View Creator to a Single View, the Aggregation ConfigMap is now automatically handled by Fast Data. The ConfigMap will be editable from the No Code Aggregation screen, without the need to access the Microservice section and the related Single View Creator. Thus, the ConfigMap will be displayed as read-only inside this section.

#### Projection changes on Mongo with Single View Trigger Generator

Starting from the `v3.0.2`, Single View Trigger Generator now supports projection changes to Mongo. Previously, the SVTG could only trigger the generation of new Single Views by sending messages onto Kafka following the `sv-trigger` format. Now, Projections Changes are generated onto the MongoDB database.

> #### Improvements

Author: ```suggestion
### Improvements
```

---
id: v11.4.0
title: Version v11.4.0 Release Notes
sidebar_label: v11.4.0
image: "img/release-note-link-preview.png"
---

_July 06th, 2023_

## Console

### New Features

### Improvements

### Bug Fix

## Fast Data

### New Features

> #### Single Views and CRUD Collections are now available as sources inside the No Code ER Schema

Reviewer: ```suggestion
#### Single Views and CRUD Collections available as sources inside the No Code ER Schema
```

Single Views and CRUD Collections can now be used as an input for the No Code ER Schema. The new sources will be available in the right panel of the canvas and can be easily added to the canvas with a simple drag-and-drop, just as it is already possible with Projections. When added to the ER Schema, Single Views and CRUD Collections are then included as available dependencies inside the No Code Aggregation.

![Collections inside ER Schema](./img/overview/collections-er-schema.png)

#### Index autogeneration for projections creates indexes for `__internal_counter` and `__internal_counterType`

The Console can now automatically generate the `__internal_counter` and `__internal_counterType` indexes, which have been introduced from `v7.5.0` of the Real-Time Updater in the update Projection queries.

> #### The Aggregation config map inside the Single View Creator microservice is automatically handled

Reviewer: ```suggestion
#### Aggregation ConfigMap automatically handled inside the Single View Creator microservice 
```

> When the attaching a Single View Creator to a Single View, the Aggregation config map is now automatically handled by Fast Data. The config map will be editable from the No Code Aggregation screen, without needing to access the Microservice section and the related Single View Creator. Thus, the config map will be displayed as read-only inside this section.

Reviewer: ```suggestion
When the attaching a Single View Creator to a Single View, the Aggregation ConfigMap is now automatically handled by Fast Data. The ConfigMap will be editable from the No Code Aggregation screen, without needing to access the Microservice section and the related Single View Creator. Thus, the ConfigMap will be displayed as read-only inside this section.
```

![Collections inside ER Schema](./img/overview/collections-er-schema.png)

#### Index autogeneration for projections creates indexes for `__internal_counter` and `__internal_counterType`

The Console can now automatically generate the `__internal_counter` and `__internal_counterType` indexes, which have been introduced from `v7.5.0` of the Real-Time Updater in the update Projection queries.

#### The Aggregation config map inside the Single View Creator microservice is automatically handled

When the attaching a Single View Creator to a Single View, the Aggregation config map is now automatically handled by Fast Data. The config map will be editable from the No Code Aggregation screen, without needing to access the Microservice section and the related Single View Creator. Thus, the config map will be displayed as read-only inside this section.

#### Projection changes on Mongo with Single View Trigger Generator

> Starting from the `v3.0.2`, Single View Trigger Generator now supports projection changes from Mongo. Previously, the SVTG could only trigger the generation of new Single Views by sending messages onto Kafka following the `sv-trigger` format. Now, Projections Changes are generated onto the MongoDB database.

Reviewer 2: ```suggestion
Starting from the `v3.0.2`, Single View Trigger Generator now supports projection changes to Mongo. Previously, the SVTG could only trigger the generation of new Single Views by sending messages onto Kafka following the `sv-trigger` format. Now, Projections Changes are generated onto the MongoDB database.
```

#### Other improvements and changes

* the UX has been improved inside the Dependency editor modal in the No Code Aggregation page with improved warnings and placeholders
* the Single View Creator column inside the Single Views page now includes links to each Single View Creator detail page instead of redirecting to the Microservices section

### Bug fix

This version addresses a bug inside the Dependency editor modal in the No Code Aggregation, which allowed the selection of Source Projections with one-to-Many condition from every configuration and not only from their parent.

Some bugs are fixed in the @mia-platform-internal/single-view-creator-lib. When Kafka fails for a faulty message, the healthiness and readiness routes now return KO; also, error logging has been fixed.

> Real-Time Updater now properly supports Debezium CDC change events with and without the change schema. In fact, _kafka-importer-lib_ has been upgraded to `v7.5.3` to fix the Debezium Message Adapter implementation.

Reviewer 2: The user does not know about the existence of the kafka importer lib, no need to mention it

Reviewer 3: You're totally right, updated!

---
id: v11.4.0
title: Version v11.4.0 Release Notes
sidebar_label: v11.4.0
image: "img/release-note-link-preview.png"
---

_July 06th, 2023_

## Console

### New Features

#### What's new section

> A big welcome to [What's new section](/development_suite/overview-dev-suite#whats-new.md), that will keep up to date with the latest product news!  
> You can access this Console section by clicking on the question mark icon located in the top bar of the Console interface.
> In case there is unread news, a notification symbol will appear so that you do not miss the latest updates that can improve your experience on Mia-Platform Console.

Reviewer: ```suggestion
A big welcome to the [What's new section](/development_suite/overview-dev-suite#whats-new.md), where you can stay up to date with the latest product news!  
You can access this section by clicking on the question mark icon located in the top bar of the Console interface.
In case there is any unread news, a notification symbol will appear, so that you do not miss the latest updates that can improve your experience on Mia-Platform Console!
```

#### Projection changes on Mongo with Single View Trigger Generator

Starting from the `v3.0.2`, Single View Trigger Generator now supports projection changes to Mongo. Previously, the SVTG could only trigger the generation of new Single Views by sending messages onto Kafka following the `sv-trigger` format. Now, Projections Changes are generated onto the MongoDB database.

#### Improvements

* the UX has been improved inside the Dependency editor modal in the No Code Aggregation page with improved warnings and placeholders
* the Single View Creator column inside the Single Views page now includes links to each Single View Creator detail page instead of redirecting to the Microservices section

### Bug Fix

This version addresses a bug inside the Dependency editor modal in the No Code Aggregation, which allowed the selection of Source Projections with one-to-Many condition from every configuration and not only from their parent.

> Some bugs are fixed in the @mia-platform-internal/single-view-creator-lib. When Kafka fails for a faulty message, the healthiness and readiness routes now return KO; also, error logging has been fixed.

Reviewer: ```suggestion
Some bugs have been fixed in the @mia-platform-internal/single-view-creator-lib. When Kafka fails for a faulty message, the healthiness and readiness routes now return KO; also, error logging has been fixed.
```

### New Features

#### Single Views and CRUD Collections are now available as sources inside the No Code ER Schema

Single Views and CRUD Collections can now be used as an input for the No Code ER Schema. The new sources will be available in the right panel of the canvas and can be easily added to the canvas with a simple drag-and-drop, just as it is already possible with Projections. When added to the ER Schema, Single Views and CRUD Collections are then included as available dependencies inside the No Code Aggregation.

![Collections inside ER Schema](./img/overview/collections-er-schema.png)

#### Index autogeneration for projections creates indexes for `__internal_counter` and `__internal_counterType`

The Console can now automatically generate the `__internal_counter` and `__internal_counterType` indexes, which have been introduced from `v7.5.0` of the Real-Time Updater in the update Projection queries.

#### The Aggregation config map inside the Single View Creator microservice is automatically handled

> When the No Code Aggregation is enabled, the Aggregation config map is automatically handled by Fast Data. The config map will be editable from the No Code Aggregation screen, without needing to access the Microservice section and the related Single View Creator. Thus, the config map will be displayed as read-only inside this section.

Reviewer 2: This is not when the no code is enabled, it is done upon connecting the svc to the sv, even if the no-code is disabled. Since from that point on you can modify the config maps from the configuration tabs in the sv/svc page

Reviewer 4: Thanks, updated
